conversion/conversion.go

```diff
@@ -1158,41 +1093,30 @@ func ProcessDump(driver string, conv *internal.Conv, r *internal.Reader) error {
 	}
 }
 
-// ProcessSchema invokes processSchema function from a sql package based on driver selected.
-func ProcessSchema(driver string, conv *internal.Conv, db *sql.DB) error {
-	switch driver {
-	case constants.MYSQL:
-		return common.ProcessSchema(conv, mysql.InfoSchemaImpl{DbName: conv.SrcDbName, Db: db})
-	case constants.POSTGRES:
-		return common.ProcessSchema(conv, postgres.InfoSchemaImpl{Db: db})
-	default:
-		return fmt.Errorf("schema conversion for driver %s not supported", driver)
-	}
-}
-
-// SetRowStats invokes SetRowStats function from a sql package based on driver selected.
-func SetRowStats(driver string, conv *internal.Conv, db *sql.DB) error {
-	switch driver {
-	case constants.MYSQL:
-		common.SetRowStats(conv, mysql.InfoSchemaImpl{DbName: conv.SrcDbName, Db: db})
-	case constants.POSTGRES:
-		common.SetRowStats(conv, postgres.InfoSchemaImpl{Db: db})
-	default:
-		return fmt.Errorf("could not set rows stats for '%s' driver", driver)
+func GetInfoSchema(driver, sqlConnectionStr string, schemaSampleSize int64) (common.InfoSchema, error) {
+	driverConfig, err := driverConfig(driver, sqlConnectionStr)
+	if err != nil {
+		return nil, err
 	}
-	return nil
-}
-
-// ProcessData invokes ProcessData function from a sql package based on driver selected.
-func ProcessData(driver string, conv *internal.Conv, db *sql.DB) error {
 	switch driver {
-	//TODO - move this logic into a factory within the sources dir
 	case constants.MYSQL:
-		common.ProcessData(conv, mysql.InfoSchemaImpl{DbName: conv.SrcDbName, Db: db})
+		db, err := sql.Open(driver, driverConfig.(string))
+		dbName := getDbNameFromSQLConnectionStr(driver, driverConfig.(string))
+		if err != nil {
+			return nil, err
+		}
+		return mysql.InfoSchemaImpl{DbName: dbName, Db: db}, nil
 	case constants.POSTGRES:
-		common.ProcessData(conv, postgres.InfoSchemaImpl{Db: db})
+		db, err := sql.Open(driver, driverConfig.(string))
+		if err != nil {
+			return nil, err
+		}
+		return postgres.InfoSchemaImpl{Db: db}, nil
+	case constants.DYNAMODB:
+		mySession := session.Must(session.NewSession())
+		dydbClient := dydb.New(mySession, driverConfig.(*aws.Config))
+		return dynamodb.InfoSchemaImpl{DynamoClient: dydbClient, SampleSize: schemaSampleSize}, nil
 	default:
-		return fmt.Errorf("data conversion for driver %s is not supported", driver)
+		return nil, fmt.Errorf("Driver %s not supported", driver)
```

Comment:
nit: error messages should not start with a capitalised letter and not end in a full stop, the lint throws a warning otherwise.

Reply:
Fixed